File: google_fetch.mjs
import * as cheerio from 'cheerio';

const queries = [
  "animatori copii Voluntari",
  "animatori copii Pipera",
  "animatori copii Iancu Nicolae"
];

const headers = {
  'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36',
  'Accept-Language': 'ro-RO,ro;q=0.9,en-US;q=0.8,en;q=0.7'
};

(async () => {
  for (const query of queries) {
    console.log(`\n--- QUERY: ${query} ---`);
    const url = `https://www.google.ro/search?q=${encodeURIComponent(query)}&gl=ro&hl=ro&num=15`;
    try {
      const response = await fetch(url, { headers });
      const html = await response.text();
      const $ = cheerio.load(html);
      
      const results = $('div.g');
      let count = 0;
      results.each((i, el) => {
        if (count >= 10) return;
        const a_tag = $(el).find('a').first();
        if (a_tag.length && a_tag.attr('href') && a_tag.attr('href').startsWith('http')) {
          const href = a_tag.attr('href');
          const title = $(el).find('h3').first().text() || "No title";
          const snippet = $(el).find('div[style="-webkit-line-clamp:2"], .VwiC3b').first().text() || "No snippet";
          console.log(`[${count+1}] TITLE: ${title}\n    URL: ${href}\n    SNIPPET: ${snippet}\n`);
          count++;
        }
      });
    } catch (e) {
      console.log(`Error fetching ${query}: ${e.message}`);
    }
    await new Promise(r => setTimeout(r, 2000));
  }
})();

File: google_fetch.py
import urllib.request
import urllib.parse
from bs4 import BeautifulSoup
import time

queries = [
    "animatori copii Voluntari",
    "animatori petreceri copii Voluntari",
    "animatori copii Pipera"
]

headers = {
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36'
}

for query in queries:
    print(f"\n--- QUERY: {query} ---")
    url = f"https://www.google.ro/search?q={urllib.parse.quote(query)}&gl=ro&hl=ro&num=15"
    try:
        req = urllib.request.Request(url, headers=headers)
        html = urllib.request.urlopen(req).read().decode('utf-8')
        soup = BeautifulSoup(html, 'html.parser')
        
        results = soup.select('div.g')
        for idx, result in enumerate(results[:10]):
            a_tag = result.select_one('a')
            if a_tag and a_tag.has_attr('href'):
                href = a_tag['href']
                title_tag = result.select_one('h3')
                title = title_tag.text if title_tag else "No title"
                snippet_tag = result.select_one('div[style="-webkit-line-clamp:2"]') or result.select_one('.VwiC3b')
                snippet = snippet_tag.text if snippet_tag else "No snippet"
                print(f"[{idx+1}] TITLE: {title}\n    URL: {href}\n    SNIPPET: {snippet}\n")
    except Exception as e:
        print(f"Error fetching {query}: {e}")
    time.sleep(2)
